### DISCUSSION TITLE
How to add third party Apps/Widgets in Canvas ?

### DISCUSSION THREAD

**Discussion by EssDevUi** [fabricjs/fabric.js#8279] [Q&A]
Hello I'm creating and editor tool using fabric js everything is working fine but now i'm stuck my client want to add third party widgets in the canvas . the client is using this for reference  https://www.fugo.ai/app/templates/e08c5313-298b-42d7-9cb0-a3dea66f8741
![image](https://user-images.githubusercontent.com/86969405/189591818-d1cfb577-92da-4a31-a597-e95b48ae8c99.png)

let me add google drive app in it . it looks like this after adding google drive app
![image](https://user-images.githubusercontent.com/86969405/189592061-7b207d15-0e1f-4e3e-98ca-105e1c59cb8d.png)

let me add Vimeo app in it  :it looks like this after adding vimeo app

![image](https://user-images.githubusercontent.com/86969405/189592211-27ad3914-6815-4036-a64f-2b14e0c52c59.png)

as you can see after embedding the apps in canvas it has their own player . i'm confused how can i achieve this functionality .

**Comment by EssDevUi** [comment]
@ShaMan123  @asturur  Any sugguestion ?

**Comment by ShaMan123** [comment]
No
#8008

**Reply by afnanz7634** [reply]
do you have any idea how can i achieve it  ? with custom canvas ?

**Reply by ShaMan123** [reply]
If I had that discussion would have been resolved and fabric would support it
as @rolitter suggests and/or http://fabricjs.com/interaction-with-objects-outside-canvas
The only browser that allows to render HTML directly onto canvas is mozilla

**Comment by rolitter** [comment]
you can create custom class,like this
https://codepen.io/pjy1711040554/pen/OJvmLgo
![image](https://user-images.githubusercontent.com/27326998/190935458-bcfe9c49-fba0-4e91-abcd-b5dd88a3fec7.png)
